### ISSUE TITLE
NullHandler isn't

### ISSUE THREAD

**Issue by jikamens** [mitsuhiko/logbook#147]
As per http://pythonhosted.org/Logbook/api/handlers.html, `NullHandler(bubble=True)` should have no effect. Alas, that is not the case. Try this:

    import logbook
    print logbook.__version__
    with logbook.NullHandler(bubble=True):
        logbook.Logger('Test').info('This message is invisible.')

If `NullHandler(bubble=True)` were working properly, it would print "This message is invisible." Alas, the message is invisible. ;-)

**Comment by vmalloc** [comment]
This is a documentation issue. We need to stop using pythonhosted.org and move to readthedocs -- I already removed that part from the documentation.

Unfortunately I need @mitsuhiko's help on this since he's the repo owner, and he's a busy guy....

**Comment by vmalloc** [comment]
The "correct" documentation is http://pythonhosted.org/Logbook/

**Comment by jikamens** [comment]
I don't think I agree that this is a documentation issue. It appears that you've merely updated the the documentation to obscure a bug in the code.

`NullHandler` has a `bubble` argument; that argument is even shown in the documentation! Since it accepts the argument, the argument should _work_. That would be the preferred behavior. But at the very least, if it's not going to work, then the handler should not accept it, and an exception should be raised if it is specified.

Allowing the argument to be specified and silently ignored is, to my mind, clearly a bug.

As for why I think it would be useful for `NullHandler` to accept and obey the `bubble` argument, I often have circumstances where I might or might not want to inject a handler into the stack, depending on the command-line options the user has specified. The cleanest way to do that is to check the options and set a handler variable to either the actual handler the user wants, or `NullHandler(bubble=True)` (if it actually worked properly), in just one place, and then to use that handler variable as a context manager everywhere else in the code. This is a reasonable and useful use case.

**Comment by vmalloc** [comment]
The "bug" is something that happened to work before, and I'm not even sure what the original use case was, or whether or not it matched your use case. IIRC the older issue opened about this by you claimed confusion with the docs, so I removed the docs that cause the confusion, because solving it would complicate the code unnecessarily, and for a use case that is not entirely clear. 

Regarding your use case, how about:

```
from contextlib import ExitStack
...
with (NullHandler() if should_mute else ExitStack()):
    pass # code here
```

I think using `bubble` for what you're describing is a bit of a corner case that is not worth the complication inside Logbook's code. `bubble` is intended for regular handlers, and `NullHandler` just happens to receive it due to inheritence.

**Comment by jikamens** [comment]
>`bubble` is intended for regular handlers, and `NullHandler` just happens to receive it due to inheritence.

So overload `__init__` for `NullHandler` so it doesn't accept `bubble`, and you're no longer misleading people.

At the very least, for heaven's sake, if `bubble` doesn't do anything from `NullHandler`, then don't include it as a valid argument in the documentation for the handler.

I wrote my own null context manager, because although I checked `contextlib` before doing so, the documentation for `ExitStack` was sufficiently obtuse that I didn't realize it would do what I needed. Thanks for the pointer, I'll use that.
